src/dissect/dissect.c

```diff
@@ -842,6 +862,31 @@ static int action_copy(DissectedImage *m, LoopDevice *d) {
                 (void) copy_times(source_fd, target_fd, 0);
 
                 /* When this is a regular file we don't copy ownership! */
+
+        } else {
+                _cleanup_strv_free_ char **list_dir = NULL;
+                size_t offset = strlen(mounted_dir);
+
+                int list_print_item(
+                                RecurseDirEvent event,
+                                const char *path,
+                                int dir_fd,
+                                int inode_fd,
+                                const struct dirent *de,
+                                const struct statx *sx,
+                                void *userdata) {
+
+                        assert_se(path);
+
+                        if (IN_SET(event, RECURSE_DIR_ENTER, RECURSE_DIR_ENTRY))
+                                printf("%s\n", path + offset);
+
+                        return RECURSE_DIR_CONTINUE;
+                }
+
+                r = recurse_dir_at(AT_FDCWD, mounted_dir, 0, UINT_MAX, RECURSE_DIR_SORT|RECURSE_DIR_ENSURE_TYPE|RECURSE_DIR_SAME_MOUNT, list_print_item, &list_dir);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to list temporary mount directory: %m");
```

Comment:
don't mention the dir is temporary, that's an implementation detail. just say "Failed to list image" or so